docs/how-tos/typescript_consume.md

```diff
@@ -0,0 +1,161 @@
+# TypeScript Consumer
+
+## Preface
+
+This guide will explain how to fetch data from a GraphQL endpoint and map the response to a TypeScript model using Relay.
+
+We will cover:
+
+- Setting up the Relay Environment in [Setup Relay Environment](#setup-relay-environment)
+- Generating the TypeScipt types with Relay in [Generate TypeScript Types](#generate-typescript-types)
+- Fetching data using Relay in [Fetch Data Using Relay](#fetch-data-using-relay)
+- Mapping the response to TypeScript types in [Mapping the response](#mapping-the-response)
+
+!!! note "Alternative Options"
+
+    If you are looking for alternative GraphQL clients, you may consider:
+
+    - [`graffle`](https://graffle.js.org/) for simpler setup
+    - [`Apollo Client`](https://www.apollographql.com/docs/react) for more features
+    - [`TypeGraphQL`](https://typegraphql.com/) if you require decorators and more advanced GraphQL server setups
+
+## Dependencies
+
+This guide assumes you have already set up Relay in your project. Relay is highly integrated with React, so you’ll also need the following packages:
+
+- `react-relay` for React integration with Relay
+- `relay-runtime` for core Relay functionality
+- `babel-plugin-relay` for compiling Relay queries
+
+## Setup Relay Environment
+
+We initially have to setup a Relay environment that helps the Relay to access the GraphQL endpoint.
+In the root of `src` we create a `RelayEnvironment.ts`. In this file we are setup up a Relay environment for making GraphQL requests.
+The fetchFn function sends a POST request to the GraphQL endpoint with the query and variables, and processes the response.
```

Comment:
Might be nice to give a quick reason for why a `fetchFn` exists - i.e. to abstract over a given networking stack and allow things like rate limiting. I'm surprised they don't provide a default one that uses `fetch` and `JSON.stringify` though

```suggestion
The `fetchFn` function sends a `POST` request to the GraphQL endpoint with the query and variables, and processes the response.
```